src/averageAges.js

```diff
@@ -36,8 +42,20 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
-function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+
+function calculateWomenAverageAge(people, withChildren = false) {
+  const female = people.filter(person => person.sex === 'f');
+  const mothers = [...new Set(people
+    .filter(person => person.mother !== null)
+    .map((person) => person.mother))];
+  const women = withChildren
+    ? female.filter(person => mothers.includes(person.name))
+    : female;
+  const ages = women
+    .map((person) => person.died - person.born)
+    .filter(age => age !== 0);
+
+  return ages.reduce((prev, curr) => prev + curr) / ages.length;
```

Comment:
you have the same reduce in the first function let's create a separate function for this logic